contracts/contracts/harvest/OETHBaseHarvester.sol

```diff
@@ -0,0 +1,234 @@
+        );
+        _;
+    }
+
+    /**
+     * @notice Verifies that the caller is either Governor or Strategist.
+     */
+    modifier onlyGovernorOrStrategistOrOperator() {
+        require(
+            msg.sender == operatorAddr ||
+                msg.sender == vault.strategistAddr() ||
+                isGovernor(),
+            "Caller is not the Operator or Strategist or Governor"
+        );
+        _;
+    }
+
+    constructor(
+        address _vault,
+        address _amoStrategy,
+        address _aero,
+        address _weth,
+        address _swapRouter
+    ) {
+        vault = IVault(_vault);
+        amoStrategy = IStrategy(_amoStrategy);
+        aero = IERC20(_aero);
+        weth = IERC20(_weth);
+        swapRouter = ISwapRouter(_swapRouter);
+    }
+
+    /**
+     * @dev Changes the operator address which can call `harvest`
+     * @param _operatorAddr New operator address
+     */
+    function setOperatorAddr(address _operatorAddr) external onlyGovernor {
+        emit OperatorChanged(operatorAddr, _operatorAddr);
+        operatorAddr = _operatorAddr;
+    }
+
+    /**
+     * @notice Collects AERO from AMO strategy and
+     *      sends it to the Strategist multisig.
+     *      Anyone can call it.
+     */
+    function harvest() external onlyGovernorOrStrategistOrOperator {
+        address strategistAddr = vault.strategistAddr();
+        require(strategistAddr != address(0), "Guardian address not set");
+
+        // Collect all AERO
+        amoStrategy.collectRewardTokens();
+
+        uint256 aeroBalance = aero.balanceOf(address(this));
+        if (aeroBalance == 0) {
+            // Do nothing if there's no AERO to transfer
+            return;
+        }
+
+        // Transfer everything to Strategist
+        aero.safeTransfer(strategistAddr, aeroBalance);
+    }
+
+    /**
+     * @notice Harvests AERO from AMO strategy and then swaps some (or all)
+     *          of it into WETH to distribute yield and fee.
+     *         When `feeBps` is set to 10000 (100%), all WETH received is
+     *          sent to strategist.
+     *
+     * @param aeroToSwap Amount of AERO to swap
+     * @param minWETHExpected Min. amount of WETH to expect
+     * @param feeBps Performance fee bps (Sent to strategist)
+     * @param sendYieldToDripper Sends yield to Dripper, if set to true.
+     *                           Otherwise, to the Guardian
+     */
+    function harvestAndSwap(
+        uint256 aeroToSwap,
+        uint256 minWETHExpected,
+        uint256 feeBps,
+        bool sendYieldToDripper
+    ) external onlyGovernorOrStrategist {
+        address strategistAddr = vault.strategistAddr();
+        require(strategistAddr != address(0), "Guardian address not set");
+
+        // Yields can only be sent to the Dripper.
+        address yieldRecipient = sendYieldToDripper
+            ? vault.dripper()
+            : strategistAddr;
+        require(yieldRecipient != address(0), "Dripper address not set");
```

Comment:
Error message might be "Yield Recipient" not set.

Reply:
Have updated it